src/main/kotlin/fi/metatavu/edufication/pages/api/controller/PagesController.kt

```diff
@@ -0,0 +1,160 @@
+package fi.metatavu.edufication.pages.api.controller
+
+import fi.metatavu.edufication.pages.api.impl.translate.ContentBlockTranslator
+import fi.metatavu.edufication.pages.api.impl.translate.PageTranslator
+import fi.metatavu.edufication.pages.api.model.ContentBlock
+import fi.metatavu.edufication.pages.api.model.PageStatus
+import fi.metatavu.edufication.pages.api.persistence.dao.ContentBlockDAO
+import fi.metatavu.edufication.pages.api.persistence.dao.PageDAO
+import fi.metatavu.edufication.pages.api.model.Page
+import java.util.*
+import javax.enterprise.context.ApplicationScoped
+import javax.inject.Inject
+import javax.transaction.Transactional
+
+/**
+ * Controller class for pages
+ */
+@ApplicationScoped
+class PagesController {
+
+    @Inject
+    private lateinit var pageDAO: PageDAO
+
+    @Inject
+    private lateinit var contentBlockDAO: ContentBlockDAO
+
+    /**
+     * Creates a new Page
+     *
+     * @param status Page status
+     * @param path Path to page
+     * @param uri Full uri path to page
+     * @param creatorId
+     * @param contentBlocks
+     *
+     * @return created counter frame
+     */
+    @Transactional
```

Comment:
This encapsulates the transaction to this specific method, you should specify it on the REST method to ensure that all db operations will be committed / rolled back on same time